### PR TITLE
Ai backends

### PR DESCRIPTION
**Pull request by go-bears**
create drop-in AI backends for OpenAI and Cohere. 

new fields

- ai.ts -where backends live
- ai.test.ts  - tiny tiny tests to structured outputs against our schema

updated files
- package.json - installed the openai and cohere packages, updated a script to run the test script

command to run the test - 4 should pass 🤞 . 
`npm run test:unit -- --reporter=verbose`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,17 +6,18 @@
     "dev": "vite dev",
     "build": "vite build",
     "preview": "vite preview",
-    "test": "npm run test:e2e && npm run test:unit -- --run",
+    "test": "npm run test:unit -- --run",
```
Comment (SybelBlue) [inline package.json:9]: why are we dropping the end-to-end tests here?

File: src/lib/ai.test.ts
Comment (SybelBlue) [inline src/lib/ai.test.ts:1]: love that this exists, thank you!

File: src/routes/api/chat/+server.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Defined in a `$REPO/.env` file as described in https://kit.svelte.dev/docs/modules#$env-static-private
 import { CHAT_API_KEY } from '$env/static/private';
 import type { Message, ChatRequest } from '$lib/types';
+import { CohereBackend, OpenAIBackend } from '$lib/ai';
```
Comment (SybelBlue) [inline src/routes/api/chat/+server.ts:4]: Is this used? (if not - delete?)

**Review by SybelBlue — COMMENTED** [review]
Had a couple questions, but it looks really good! We're waiting on #16, regardless.

File: package.json (hunk 1 above)
Comment (go-bears) [reply package.json:9]: explained in person. the e2e webserver was getting in the way of running the unit test. disabled to confirm backend works

File: src/routes/api/chat/+server.ts (hunk 2 above)
Comment (go-bears) [reply src/routes/api/chat/+server.ts:4]: sure thing! I was stubbing out the imports for the refeactoring step of pulling out the hardcoded call. I'm totally with good with stripping it out here and adding it back in. Agree that adding it in this pr isn't pretty / neat.

**Pull request merged**